modules/nlu/src/views/entities/editor.jsx

```diff
@@ -0,0 +1,168 @@
+
+    this.setState({ currentEntity: entity }, this.onUpdate)
+  }
+
+  removeSynonym = (occurence, index) => {
+    let entity = this.state.currentEntity
+
+    occurence.synonyms.splice(index, 1)
+
+    const eIndex = entity.occurences.findIndex(o => o.name === occurence.name)
+    entity.occurences[eIndex] = occurence
+    this.setState({ currentEntity: entity }, this.onUpdate)
+  }
+
+  onSynonymInputChange = event => {
+    const value = event.target.value
+    this.setState({ synonymInput: value })
+  }
+
+  addOccurence = () => {
+    const occurence = this.state.occurenceInput
+    let entity = this.state.currentEntity
+
+    if (entity.occurences.includes(occurence)) {
+      return
+    }
+
+    entity.occurences = [...entity.occurences, { name: occurence, synonyms: [] }]
+
+    this.occurenceInputRef.current.value = ''
+    this.setState({ currentEntity: entity, occurenceInput: '' }, this.onUpdate)
+  }
+
+  handleOccurenceEnter = event => {
+    this.onEnterPressed(event, this.addOccurence)
+  }
+
+  onOccurenceInputChange = event => {
+    const value = event.target.value
+    this.setState({ occurenceInput: value })
+  }
+
+  selectOccurence = occurence => {
+    if (occurence !== this.state.currentOccurence) {
+      this.setState({ currentOccurence: occurence })
+    }
+  }
+
+  removeOccurence = occurence => {
+    const confirm = window.confirm(
+      `Are you sure you want to delete the occurence "${occurence.name}" and all its synonyms?`
+    )
+    if (!confirm) {
+      return
+    }
+
+    const entity = this.state.currentEntity
+    const index = entity.occurences.findIndex(o => o.name === occurence.name)
+
+    entity.occurences.splice(index, 1)
+    this.setState({ currentEntity: entity, currentOccurence: entity.occurences[0] }, this.onUpdate)
+  }
+
+  renderOccurences = () => {
+    const occurences = this.state.currentEntity && this.state.currentEntity.occurences
+    if (!occurences) {
+      return null
+    }
+
+    return (
+      <div>
+        <input
+          class={classNames('form-control', style.occurenceInput)}
+          ref={this.occurenceInputRef}
+          type="text"
+          placeholder="Type to create an occurence"
+          onKeyDown={this.handleOccurenceEnter}
+          onChange={this.onOccurenceInputChange}
+        />
+        {occurences.map(o => (
+          <ListGroupItem className={style.occurence}>
+            <div className={style.occurenceName}>{o.name}</div>
+            <ReactTags
+              placeholder="Enter a synonym"
+              tags={o.synonyms}
+              handleDelete={index => this.removeSynonym(o, index)}
+              handleAddition={e => this.addSynonym(o, e)}
+              allowDeleteFromEmptyInput={false}
+            />
+            <Glyphicon glyph="trash" className={style.occurenceDelete} onClick={() => this.removeOccurence(o)} />
+          </ListGroupItem>
+        ))}
+      </div>
+    )
+  }
+
+  onUpdate = () => {
```

Comment:
Since this method is used to update the parent, name should be more like updateData or something similar, since it's not an 'event'